frontend/src/components/PostsPagination.js

```diff
@@ -23,25 +21,25 @@ const PostsPagination = ({ posts }) => {
   }, [posts.data.length])
 
   const switchToSpecificPage = page => {
-    if (!history.location.search && page === 1) return
+    if (!router.location.search && page === 1) return
     if (page !== DOTS)
-      pathname.includes('category')
-        ? history.push(`${pathname}?page=${page}`)
-        : history.push(`/?page=${page}`)
+      router.pathname.includes('category')
+        ? router.push(`${router.pathname}?page=${page}`)
```

Comment:
```suggestion
        ? router.push(`${router.asPath}?page=${page}`)
```

pleas check all the places where pathname is used